source/git-promise.js

```diff
@@ -65,9 +66,10 @@ const gitExecutorProm = (args, retryCount) => {
         };
         const gitProcess = child_process.spawn(gitBin, args.commands, procOpts);
         if (args.timeout) {
-          setTimeout(() => {
+          timeoutTimer = setTimeout(() => {
             if (isCompleted) return;
```

Comment:
As far as I can see `isCompleted` is only used to check if the timeout callback has been called or is still required, which seems like a very similar use case to the new `timeoutTimer`.

Can you replace checking `isCompleted` with checking if `timeoutTimer` is not `null`?

Reply:
Yes, it looks like it can be changed as suggested because we always set a timeout:
https://github.com/FredrikNoren/ungit/blob/70eb8246421658358d3ef08cfbe6a1974bd9b3b9/source/git-promise.js#L159

Not sure why the `if (args.timeout)` is there.